### PR TITLE
wgsl: Add extractBits, insertBits

### PR DESCRIPTION
**Pull request by dneto0**
Fixed: #2129 #288

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -9346,6 +9346,60 @@ struct __modf_result_vecN {
...
+    <li>The result is 0 if |c| is 0.
+    <li>Otherwise, bits 0..|c|-1 of the result are copied from bits
+       |o|..|o|+|c|-1 of |e|.
+       Other bits of the result are 0.
+    </ul>
+    [=Component-wise=] when |T| is a vector.
+    <br>(OpBitFieldUExtract)
+
+  <tr algorithm="insert bits">
+    <td>|T| is [UNSIGNEDINTEGRAL]
```
Comment (alan-baker) [inline wgsl/index.bs:9386]: Shouldn't `T` be integral (both u32 and i32)?

**Review by alan-baker — CHANGES_REQUESTED** [review]
The rendering of the table isn't great. The description column for the whole table gets squeezed by the long overload that can't wrap. Can you manually add some breaks (maybe parameters on different lines) to reduce the required width?

File: wgsl/index.bs (hunk 1 above)
Comment (dneto0) [reply wgsl/index.bs]: fixed.

**Review by alan-baker — APPROVED** [review]
The table is much improved, thanks!

**Review by kvark — CHANGES_REQUESTED** [review]
waiting for https://github.com/gpuweb/gpuweb/issues/2129#issuecomment-1004324101 resolution

**Review by kvark — APPROVED** [review]

**Comment by dneto0** [comment]
Rebased and resolved trivial conflicts.

**Comment by dneto0** [comment]
FYI for implementors.

There is a relatively simple polyfill for extractBits that works for both signed and unsigned variants: https://github.com/gpuweb/gpuweb/issues/2129#issuecomment-1010267624

**Comment by kdashg** [comment]
<details><summary><a href="https://github.com/gpuweb/gpuweb/wiki/WGSL-2022-01-11-Minutes#wgsl-add-extractbits-insertbits-2466">WGSL meeting minutes 2022-01-11</a></summary>



* KG: And we can polyfill this? (msl &lt;1.2)
* DM: Yes.
* DN: The tricky one was signed case for extractbits.  [https://github.com/gpuweb/gpuweb/issues/2129#issuecomment-1010267624](https://github.com/gpuweb/gpuweb/issues/2129#issuecomment-1010267624) 
* MM: Care mainly that the polyfill is branchless.
* DN: Yes, it was simpler than I thought, and branchless.
* **Approved**


</details>

**Pull request merged**